testdata/schema.go

```diff
@@ -0,0 +1,232 @@
+	basicFields = append(basicFields, arrow.Field{Name: "inet", Type: types.NewInetType(), Nullable: true})
+	basicFields = append(basicFields, arrow.Field{Name: "mac", Type: types.NewMacType(), Nullable: true})
+
+	// sort and remove duplicates (e.g. date32 and date64 appear twice)
+	basicFields = sortAndRemoveDuplicates(basicFields)
+
+	// we don't support float16 right now
+	basicFields = removeFieldsByType(basicFields, arrow.FLOAT16)
+
+	if !opts.IncludeTimestamps {
+		// for backwards-compatibility, microsecond timestamps are not removed here
+		basicFields = removeFieldsByDataType(basicFields, &arrow.TimestampType{Unit: arrow.Second, TimeZone: "UTC"})
+		basicFields = removeFieldsByDataType(basicFields, &arrow.TimestampType{Unit: arrow.Millisecond, TimeZone: "UTC"})
+		basicFields = removeFieldsByDataType(basicFields, &arrow.TimestampType{Unit: arrow.Nanosecond, TimeZone: "UTC"})
+	}
+	if !opts.IncludeDates {
+		basicFields = removeFieldsByType(basicFields, arrow.DATE32, arrow.DATE64)
+	}
+	if !opts.IncludeTimes {
+		basicFields = removeFieldsByType(basicFields, arrow.TIME32, arrow.TIME64)
+	}
+	if !opts.IncludeIntervals {
+		basicFields = removeFieldsByType(basicFields, arrow.INTERVAL_DAY_TIME, arrow.INTERVAL_MONTHS, arrow.INTERVAL_MONTH_DAY_NANO)
+	}
+	if !opts.IncludeDurations {
+		basicFields = removeFieldsByType(basicFields, arrow.DURATION)
+	}
+	if !opts.IncludeLargeTypes {
+		basicFields = removeFieldsByType(basicFields, arrow.LARGE_BINARY, arrow.LARGE_STRING)
+	}
+
+	var compositeFields []arrow.Field
+
+	// we don't need to include lists of binary or large binary right now; probably no destinations or sources need to support that
+	basicFieldsWithExclusions := removeFieldsByType(basicFields, arrow.BINARY, arrow.LARGE_BINARY)
+	if opts.IncludeLists {
+		compositeFields = append(compositeFields, ListOfFields(basicFieldsWithExclusions)...)
+	} else {
+		// only include lists that were originally supported by CQTypes
+		cqListFields := []arrow.Field{
+			{Name: "string", Type: arrow.BinaryTypes.String, Nullable: true},
+			{Name: "uuid", Type: types.NewUUIDType(), Nullable: true},
+			{Name: "inet", Type: types.NewInetType(), Nullable: true},
+			{Name: "mac", Type: types.NewMacType(), Nullable: true},
+		}
+		compositeFields = append(compositeFields, ListOfFields(cqListFields)...)
+	}
+
+	if opts.IncludeMaps {
+		compositeFields = append(compositeFields, MapOfFields(basicFieldsWithExclusions)...)
+	}
+
+	// add JSON later, we don't want to include it as a list or map right now (it causes complications with JSON unmarshalling)
+	basicFields = append(basicFields, arrow.Field{Name: "json", Type: types.NewJSONType(), Nullable: true})
+
+	if opts.IncludeStructs {
+		// struct with all the types
+		compositeFields = append(compositeFields, arrow.Field{Name: "struct", Type: arrow.StructOf(basicFields...), Nullable: true})
+
+		// struct with nested struct
```

Comment:
+ with field as list of steucts, as map with struct as field

Reply:
We can always add more cases later, I tried to keep it relatively simple for now